bayes_kit/ess.py

```diff
@@ -92,9 +92,9 @@ def ess_ipse(chain: VectorType) -> FloatType:
         raise ValueError(f"ess requires len(chains) >=4, but {len(chain) = }")
     acor = autocorr(chain)
     n = first_neg_pair_start(acor)
-    sigma_sq_hat = acor[0] + 2 * sum(acor[1:n])
+    sigma_sq_hat = acor[0] + 2 * acor[1:n].sum()
     ess = len(chain) / sigma_sq_hat
-    return ess
+    return np.float64(ess)
```

Comment:
I think these functions should return a normal python `float` rather than np.float64, which will also avoid this cast

Reply:
The existence of two 64-bit float and two 64-bit int types is driving me bonkers.  Is there anything you can recommend reading to understand the difference between `float` and `np.float64`?  Specifically, why would anyone use a 64-bit floating point representation that's not IEEE 754 compliant?  And what's the difference between `int` and `np.int32` and `np.int64`?

Reply:
Oh, and I made the fix for this.

Reply:
`float` in Python doesn’t have a very specific specification other than the note that they “are usually implemented using double in C; information about the precision and internal representation of floating point numbers for the machine on which your program is running is available in [sys.float_info](https://docs.python.org/3/library/sys.html#sys.float_info)” (from https://docs.python.org/3/library/stdtypes.html#typesnumeric). Meanwhile, numpy’s `np.float64` is just an alias of `np.double`, which is a c double, so using CPython these will always be the same underlying representation and precision. 

In general I prefer to use the python version - numpy is good about accepting it, but not everything else is good about accepting numpy’s. 



Integer types are a whole other can of worms. `int` in python is what most other languages would call a Big Integer - it is an unlimited precision number. Meanwhile, numpy’s types are all roughly equivalent to the similarly named C99 typedefs like `int64_t`